my-app/src/components/championsGrid/ChampionsGrid.tsx

```diff
@@ -4,27 +4,61 @@ import { useSelector } from "react-redux";
 import fetchApi from "../../features/champions/api/fetchApi";
 import { useDispatch } from "react-redux";
 import { useEffect } from "react";
-import FetchButton from "../../components/Button/button";
 import { ToastContainer } from "react-toastify";
 import "react-toastify/dist/ReactToastify.css";
 import { useNavigate } from "react-router-dom";
-import ChampionModal from "../PopupModal/championModal";
-
-import "./PaginatedItems.scss";
-import saveToLocalInstance from "../../components/WatchList/watchlist-save-local";
+import ChampionModal from "../championModal";
+import "./championsGrid.scss";
+import CheckSaved from "../watchlist/checkStatus";
+import Champion from "../../features/champions/types/champion";
 
 const PaginatedItems = (props: any) => {
+	const dispatch = useDispatch();
 	const navigate = useNavigate();
 	const [show, setShow] = React.useState(false);
-
 	const [modalChampion, setModalChampion] = React.useState(0);
 	const path = window.location.pathname;
-	const state = useSelector((state: any) => state);
-	const dispatch = useDispatch();
-	const { loading, error, champions, totalPages } = state.champions;
-
+	const reduxState = useSelector((state: any) => state.champions);
+	const { loading, error, champions, totalPages } = reduxState;
 	const pageNumber = path.match(/\d+/g);
 	const page = pageNumber ? pageNumber[0] : 1;
+	const [championsSort, setChampionsSort] = React.useState<Champion[]>([
+		...champions,
+	]);
+
+	if (champions && champions.length > 0 && championsSort.length === 0) {
+		setChampionsSort(champions);
+	}
+
+	const sortButtons = () => {
+		return (
+			<div className="sort-buttons">
+				<button className="sort-button">
+					<i className="fas fa-sort-amount-up" onClick={() => sort("ASC")}></i>
+				</button>
+				<button className="sort-button">
+					<i
+						className="fas fa-sort-amount-down"
+						onClick={() => sort("DESC")}
+					></i>
+				</button>
+			</div>
+		);
+	};
+
+	const sort = (order: string) => {
+		order === "ASC"
+			? setChampionsSort([
+					...championsSort.sort((a: Champion, b: Champion) => {
+						return a.armor > b.armor ? -1 : b.armor > a.armor ? 1 : 0;
+					}),
+			  ])
+			: setChampionsSort([
+					...championsSort.sort((a: Champion, b: Champion) => {
+						return a.armor > b.armor ? 1 : b.armor > a.armor ? -1 : 0;
+					}),
+			  ]);
+	};
```

Comment:
I think this can be safely moved into a redux reducer function which would be triggered by a nice redux action like:
```js
// actions.ts
enum SortDirection {
 ASC = "ASC",
 DESC = "DESC",
}
const sort = (direction: SortDirection) => { type: SORT, direction };
```

Reply:
Done